src/middleware/mod.rs

```diff
@@ -178,6 +178,60 @@ impl Chain {
     }
 }
 
+/// Builder struct for a `Chain`
+///
+pub struct ChainLink {
```

Comment:
I'd probably use a wrapper struct like `ChainLink(Chain)`, because the name of the field of your struct doesn't really add any information.